include/internal/catch_decomposer.h

```diff
@@ -142,7 +142,17 @@ namespace Catch {
         auto operator <= ( RhsT const& rhs ) -> BinaryExpr<LhsT, RhsT const&> const {
             return { static_cast<bool>(m_lhs <= rhs), m_lhs, "<=", rhs };
         }
+        
+        template<typename RhsT>
+        void operator&&( RhsT const&) = delete;
+
+        void operator&&(bool) = delete;
```

Comment:
Why use an explicit specialization for `bool`?

Reply:
I am not entirely sure about this but if we have something like
`REQUIRE(Factorial(1) == 1  && Factorial(2)==2);`
then the second operand after `&&` would be resolved as a `bool`?
And I think this why the original code specialized it for `bool` too.
`auto operator!=(bool rhs) -> BinaryExpr<LhsT, bool> const {
            return {m_lhs != rhs, m_lhs, "!=", rhs};
        }`
Just a way to tell that chaining with `&&` is not acceptable or something...